### PR TITLE
Fix controller manager to only accept args from command line

### PR DESCRIPTION
**Pull request by marun**
As per a [PR comment](https://github.com/kubernetes-sigs/federation-v2/pull/216#discussion_r212668792) from @pmorie, the controller manager should accept arguments set via the downward api from the command line rather than as environment variables.

### CODE REVIEW

File: scripts/generate-namespaced-install-yaml.sh [hunk 1]
```diff
@@ -32,11 +32,14 @@ sed -i -e '/^  namespace: federation-system$/d' "${INSTALL_YAML}"
 # Convert rbac from cluster- to namespace-scoped
 sed -i -e 's/ClusterRole/Role/' "${INSTALL_YAML}"
 
-# Add --limited-scope arg to container
-sed -i -e 's/\(\s*- \)\(--install-crds=false\)/\1\2\n\1--limited-scope=true/' "${INSTALL_YAML}"
+# Add args to container
+sed -i -e '/--install-crds=false/a\
+        - --federation-namespace=FEDERATION_NAMESPACE\
```
Comment (pmorie) [inline scripts/generate-namespaced-install-yaml.sh]: Should this be `$FEDERATION_NAMESPACE` ?

File: scripts/generate-namespaced-install-yaml.sh [hunk 2]
```diff
@@ -32,11 +32,15 @@ sed -i -e '/^  namespace: federation-system$/d' "${INSTALL_YAML}"
...
+sed -i -e '/--install-crds=false/a\
+        - --limited-scope=true\
+        - --federation-namespace=$(FEDERATION_NAMESPACE)\
+        - --registry-namespace=$(CLUSTER_REGISTRY_NAMESPACE)' "${INSTALL_YAML}"
 
 # Add namespace env args to container
- sed -i -e '/terminationGracePeriodSeconds/i\        env:\
+ sed -i -e '/terminationGracePeriodSeconds/i\
+        env:\
         - name: FEDERATION_NAMESPACE\
```
Comment (gyliu513) [inline scripts/generate-namespaced-install-yaml.sh]: Why still update env for `FEDERATION_NAMESPACE` and `CLUSTER_REGISTRY_NAMESPACE` to `install-namespaced.yaml`?

File: scripts/generate-namespaced-install-yaml.sh [hunk 3]
```diff
@@ -32,11 +32,15 @@ sed -i -e '/^  namespace: federation-system$/d' "${INSTALL_YAML}"
 # Convert rbac from cluster- to namespace-scoped
 sed -i -e 's/ClusterRole/Role/' "${INSTALL_YAML}"
 
-# Add --limited-scope arg to container
-sed -i -e 's/\(\s*- \)\(--install-crds=false\)/\1\2\n\1--limited-scope=true/' "${INSTALL_YAML}"
+# Add args to container
+sed -i -e '/--install-crds=false/a\
+        - --limited-scope=true\
+        - --federation-namespace=$(FEDERATION_NAMESPACE)\
```
Comment (gyliu513) [inline scripts/generate-namespaced-install-yaml.sh]: I think https://github.com/kubernetes-sigs/federation-v2/blob/master/scripts/deploy-federation.sh#L130 also need some update to set those two variables:  `FEDERATION_NAMESPACE` and `CLUSTER_REGISTRY_NAMESPACE `.

File: scripts/generate-namespaced-install-yaml.sh (hunk 2 above)
Comment (marun) [reply scripts/generate-namespaced-install-yaml.sh]: The goal is setting these environment variables to the namespace the pod is running in (i.e. via the downward api).

Reference: https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#use-pod-fields-as-values-for-environment-variables

File: scripts/generate-namespaced-install-yaml.sh (hunk 3 above)
Comment (marun) [reply scripts/generate-namespaced-install-yaml.sh]: These variables are being set from the pod's namespace, and the pod's namespace is determined by the [namespace the deployment script applies the yaml to](https://github.com/kubernetes-sigs/federation-v2/blob/master/scripts/deploy-federation.sh#L139).

Comment (gyliu513) [reply scripts/generate-namespaced-install-yaml.sh]: I see, thanks @marun

**Pull request merged**